### PR TITLE
rpc: add getblockfrompeer RPCTypeCheck and invalid input test coverage

### LINKED ISSUE #24943: RPC `getblockfrompeer` returns an error with bitcoin-cli
I encounter an error with new command `getblockfrompeer`.

```
bitcoin-cli getblockfrompeer "000000000000034a7dedef4a161fa058a2d67a173a90155f3a2fe6fc132e0ebf" 0
error code: -1
error message:
JSON value is not an integer as expected
 ```

It's working with curl:
```
curl -v --user myusername --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "getblockfrompeer", "params": ["000000000000034a7dedef4a161fa058a2d67a173a90155f3a2fe6fc132e0ebf", 0] }' -H 'content-type: text/plain;' http://127.0.0.1:8332/
```

I use the version bitcoin-core-23.0/test.rc5/bitcoin-23.0rc5-arm64-apple-darwin.dmg

### PR DESCRIPTION
**Pull request by jonatack**
The new getblockfrompeer RPC lacks test coverage for invalid arguments, and its error messages are not harmonized with the existing RPCs.

Fix all issues.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
> Closes https://github.com/bitcoin/bitcoin/issues/24943.

Pretty sure the changes here have nothing to do with the issue. The issue report is about bitcoin-cli. Here, you are modifying the server code.

**Comment by jonatack** [comment]
Yes. Removed the mention in the description.

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -439,6 +439,11 @@ static RPCHelpMan getblockfrompeer()
         },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
 {
+    RPCTypeCheck(request.params, {
+        UniValue::VSTR, // blockhash
+        UniValue::VNUM, // peer_id
+    }, true);
```
[the remark this answers is not in the document]
Comment (jonatack) [reply src/rpc/blockchain.cpp:445]: It already throws with the error seen in the previously referenced issue, but we do the same standard type check and user-friendly error message in many RPCs:

scantxoutset
sendrawtransaction
testmempoolaccept
estimatesmartfee
estimaterawfee
getdescriptorinfo
deriveaddresses
setmocktime
mockscheduler
addconnection
createrawtransaction
decodescript
signrawtransactionwithkey
decodepsbt
combinepsbt
finalizepsbt
createpsbt
converttopsbt
utxoupdatepsbt
joinpsbts
analyzepsbt
upgradewallet
importmulti
importdescriptors
lockunspent
getunconfirmedbalance
walletcreatefundedpsbt
walletprocesspsbt
send
fundrawtransaction
bumpfee
psbtbumpfee

...and the new sendall RPC.

So it seems correct to do it here as well. If you disagree, can you clarify further?

[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/rpc/blockchain.cpp:445]: Yeah, I think it should be removed in the other RPCs as well.

[the remark this answers is not in the document]
Comment (jonatack) [reply src/rpc/blockchain.cpp:445]: Ok, that seems to be a discussion for another pull. This one adds test coverage and harmonizes getblockfrompeer with the other RPCs listed. LMK if you'd prefer the second commit be dropped.

**Review by luke-jr — APPROVED** [review]
utACK

File: src/rpc/blockchain.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/rpc/blockchain.cpp:445]: `gettxout` is a "real" RPC and isn't checked either, so I don't see why this one should be checked for "harmony" as well.

Recall that this is a test-only RPC, so any effort spent on it won't be visible by users.

[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/rpc/blockchain.cpp:445]: Started my suggestion in #25093, which should be trivial to extend to the type checking you want to add here.

Comment (w0xlt) [inline src/rpc/blockchain.cpp:445]: nit: This RPC does not  accept null params.

```suggestion
    RPCTypeCheck(request.params, {
        UniValue::VSTR, // blockhash
        UniValue::VNUM, // peer_id
    });
```

**Review by w0xlt — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/24944/commits/a926025ca822ae1fe480c6b1a2c480d1d4641c06

**Comment by jonatack** [comment]
Thanks @luke-jr and @w0xlt for the ACKs, updated per `git diff a926025 2ef5294` to take @w0xlt's suggestion.

**Review by brunoerg — APPROVED** [review]
ACK 2ef5294a5bb68ceb3797d2638567a172cc21699f

**Comment by jonatack** [comment]
This PR had 2 ACKs for a couple months and now another one for a simple and non-risky change. Anything further needed here?

**Comment by MarcoFalke** [comment]
edited comment (https://github.com/bitcoin/bitcoin/pull/24944#issuecomment-1167214427) due to a bug or feature (?) in github-merge.py:

```
Merge message contains an @!
---------------------------------------------------------------------------
Merge bitcoin/bitcoin#24944: rpc: add getblockfrompeer RPCTypeCheck and invalid input test coverage

2ef5294a5bb68ceb3797d2638567a172cc21699f rpc: add RPCTypeCheck for getblockfrompeer inputs (Jon Atack)
734b9669ff7b2f5e2820993443a6f868f6b0b20a test: add getblockfrompeer coverage of invalid inputs (Jon Atack)

Pull request description:

  The new getblockfrompeer RPC lacks test coverage for invalid arguments, and its error messages are not harmonized with the existing RPCs.
  
  Fix all issues.


ACKs for top commit:
  jonatack:
    Thanks @luke-jr and @w0xlt for the ACKs, updated per `git diff a926025 2ef5294` to take @w0xlt's suggestion.
  brunoerg:
    ACK 2ef5294a5bb68ceb3797d2638567a172cc21699f

**Comment by MarcoFalke** [comment]
My thinking is that we should make `RPCTypeCheck` to be run internally and consistently on all methods, not intermittently on only those that it was added to. See also https://github.com/bitcoin/bitcoin/pull/24944#discussion_r868095977

**Comment by MarcoFalke** [comment]
So I am ~0, but happy to merge if reviewers think it is useful.

**Pull request merged**